public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_resource_graph/azure_resource_graph_datasource.test.ts

```diff
@@ -0,0 +1,62 @@
+import { TemplateSrv } from 'app/features/templating/template_srv';
+import { backendSrv } from 'app/core/services/backend_srv';
+import AzureResourceGraphDatasource from './azure_resource_graph_datasource';
+
+const templateSrv = new TemplateSrv();
+
+jest.mock('app/core/services/backend_srv');
+jest.mock('@grafana/runtime', () => ({
+  ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+  getBackendSrv: () => backendSrv,
+  getTemplateSrv: () => templateSrv,
+}));
+
+describe('AzureResourceGraphDatasource', () => {
+  const datasourceRequestMock = jest.spyOn(backendSrv, 'datasourceRequest');
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    datasourceRequestMock.mockImplementation(jest.fn());
+  });
+
+  const ctx: any = {};
+
+  beforeEach(() => {
+    ctx.instanceSettings = {
+      url: 'http://azureresourcegraphapi',
+    };
+
+    ctx.ds = new AzureResourceGraphDatasource(ctx.instanceSettings);
+  });
+
+  describe('When applying template variables', () => {
```

Comment:
I'm not entirely sure what this is testing. The point of the `applyTemplateVariables` function is to expand variables and store it on the query model before sending it to the backend.